microsoftdynamicscrm/metadata_test.go

```diff
@@ -0,0 +1,217 @@
+		expectedErrs        []error
+	}{
+		{
+			name:  "At least one object name must be queried",
+			input: nil,
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusTeapot)
+			})),
+			expectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			name:  "Mime response header expected",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusTeapot)
+			})),
+			expectedErrs: []error{interpreter.ErrMissingContentType},
+		},
+		{
+			name:  "Missing XML response on status OK",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				writeBody(w, "")
+			})),
+			expectedErrs: []error{common.ErrNotXML},
+		},
+		{
+			name:  "Missing XML root",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				writeBody(w, `<?xml version="1.0" encoding="utf-8"?>`)
+			})),
+			expectedErrs: []error{common.ErrNoXMLRoot},
+		},
+		{
+			name:  "Server response without CRM Schema",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				writeBody(w, `
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx"></edmx:Edmx>`)
+			})),
+			expectedErrs: []error{ErrMissingSchema},
+		},
+		{
+			name:  "Object name cannot be found from server response",
+			input: []string{"msfp_surveyinvite"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write(fakeServerResp)
+			})),
+			expectedErrs: []error{ErrObjectNotFound, errors.New("unknown entity msfp_surveyinvite")}, // nolint:goerr113
+		},
+		{
+			name:  "Correctly list metadata for account leads and invite contact",
+			input: []string{"accountleads", "adx_invitation_invitecontacts"},
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/xml")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write(fakeServerResp)
+			})),
+			expected: &common.ListObjectMetadataResult{
+				Result: map[string]common.ObjectMetadata{
+					"accountleads": {
+						DisplayName: "accountleads",
+						FieldsMap: map[string]string{
+							"accountid":                 "accountid",
```

Comment:
So Microsoft Dynamics doesn't return the display name for objects or fields? What I mean is: for the object `accountleads`, does the API not return something like `Account Leads`?

Reply:
Yes, indeed I couldn't find the human readable forms in either Endpoint response nor in Schema response.
https://github.com/Cobalt0s/connectors/blob/1cbb441a9b93a82ba1b7280a471b895e49d1151b/microsoftdynamicscrm/test/metadata.xml#L14-L37
Some screenshot I took earlier. All responses look quite "unnatural" for display.
![image](https://github.com/amp-labs/connectors/assets/60330780/b92454e9-513a-4dc9-ae60-59de96a8bd27)

Reply:
that's unfortunate 😢